src/main/java/com/prajina/academy/service/CourseServiceImpl.java

```diff
@@ -1,142 +1,88 @@
 package com.prajina.academy.service;
 
-import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.atomic.AtomicLong;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import com.prajina.academy.dao.CourseDAO;
 import com.prajina.academy.model.Course;
+import com.prajina.academy.transformer.CourseMapper;
 
 @Service("CourseService")
```

Comment:
you need the @Service annotation, but you don't need to give it a string "CourseService"  unless you have multiple bean candidates for CourseService.